deploy/job-manager/deploy.sh

```diff
@@ -0,0 +1,329 @@
+    kubectl apply -f job-manager-service.yaml
+}
+
+# Tested
+function apply_kube_ingress() {
+    local ENV=$1
+    local EXTERNAL_IP_NAME="job-manager"
+    local TLS_SECRET_NAME="${ENV}-mint-ssl"
+    # local VAULT_TOKEN_FILE
+
+    stdout "Rendering job-manager-ingress.yaml file"
+    docker run -i --rm \
+        -e EXTERNAL_IP_NAME=${EXTERNAL_IP_NAME} \
+        -e TLS_SECRET_NAME=${TLS_SECRET_NAME} \
+        -v ${PWD}:/working broadinstitute/dsde-toolbox:k8s \
+        /usr/local/bin/render-ctmpl.sh -k /working/job-manager-ingress.yaml.ctmpl
+
+    stdout "Applying job-manager-ingress.yaml"
+    kubectl apply -f job-manager-ingress.yaml
+}
+
+function tear_down_kube_secret() {
+    local filename=$1
+
+    stdout "Tearing down secret objects on Kubernetes cluster"
+    kubectl delete secret ${filename}
+}
+
+function tear_down_kube_configMap() {
+    local filename=$1
+
+    stdout "Tearing down configMap objects on Kubernetes cluster"
+    kubectl delete configmap ${filename}
+}
+
+function tear_down_rendered_files() {
+
+    stdout "Removing all generated files"
+    rm -rf ".htpasswd"
+    rm -rf "api-config.json"
+    rm -rf "environment.template.ts"
+    rm -rf "job-manager-deployment.yaml"
+    rm -rf "job-manager-ingress.yaml"
+    rm -rf "nginx.conf"
+}
+
+# The main function to execute all steps of a deployment of Job Manager
+function main() {
+    ENV=$1
+    JM_TAG=$2
+    JMUI_USR=$3
+    JMUI_PWD=$4
+    VAULT_TOKEN_FILE=${5:-"$HOME/.vault-token"}
+
+    DOCKER_TAG=${JM_TAG}
+    API_DOCKER_IMAGE="gcr.io/broad-dsde-mint-${ENV}/jm-cromwell-api:${DOCKER_TAG}"
+    UI_DOCKER_IMAGE="gcr.io/broad-dsde-mint-${ENV}/jm-cromwell-ui:${DOCKER_TAG}"
+
+    set -e
+
+    line
+    configure_mint_kubernetes ${ENV}
+
+    line
+    update_submodule ${JM_TAG}
+
+    line
+    render_environment_ts
+
+    line
+    inject_angular_and_build_UI ${ENV} ${DOCKER_TAG}
+
+    line
+    build_API ${ENV} ${DOCKER_TAG}
+
+    api_config="cromwell-credentials-$(date '+%Y-%m-%d-%H-%M')"
+#    create_API_config ${ENV} ${api_config} ${VAULT_TOKEN_FILE}
+
+    capabilities_config="capabilities-config-$(date '+%Y-%m-%d-%H-%M')"
+#    create_API_capabilities_conf ${capabilities_config}
+
+    username=${JMUI_USR}
+    password=${JMUI_PWD}
+    ui_proxy="jm-htpasswd-$(date '+%Y-%m-%d-%H-%M')"
+#    create_UI_proxy ${username} ${password} ${ui_proxy}
+
+    ui_config="jm-ui-config-$(date '+%Y-%m-%d-%H-%M')"
+#    create_UI_conf ${ui_config}
+
+    line
+    if create_API_config ${ENV} ${api_config} ${VAULT_TOKEN_FILE} && create_API_capabilities_conf ${capabilities_config} && create_UI_proxy ${username} ${password} ${ui_proxy} && create_UI_conf ${ui_config} ${JM_TAG}
+    then
+        stdout "Successfully created all config files on Kubernetes cluster"
+    else
+        tear_down_kube_secret ${api_config} || tear_down_kube_secret ${ui_proxy} || tear_down_kube_configMap ${capabilities_config} || tear_down_kube_configMap ${ui_config}
```

Comment:
If we cannot create all of the config files, should this tear down all of them? Or is there a way to check after trying to create each file?